### PR TITLE
[Content] NLCD Dataset Landing Page

### PR DESCRIPTION
**Pull request by acblackford**
## Why are you creating this Pull Request?

- [Adding Datasets or Stories](?title=Content%3A%20%3Cname%3E&expand=1&template=content.md)
- [Version Release](?title=Deploy%20vX.X.X&expand=1&template=version_release.md)
- [Other](?expand=1&template=default.md)

### CODE REVIEW

File: datasets/nlcd.data.mdx
Comment (slesaad) [inline datasets/nlcd.data.mdx:1]:
```suggestion
---
id: nlcd
```

**Comment by jerikac** [comment]
@slesaad can we get a final review before merging to main? from what I see things are great! Is there anyone on the UI/front end that needs to review before pushing this to production?

**Comment by anayeaye** [comment]
Are these datasets ready for production? We have an automated promotion step coming soon but today we will still need to publish to openveda.cloud/api/stac. Also is there a copy of the json used to publish these collections to staging? We can reverse engineer it, too

- `nlcd-new-urbanization`
- `nlcd-annual-conus`

cc: @jerikac @acblackford  @slesaad

Also: please disregard if you all are already in touch about this, I just noticed the PR with staging data!

**Comment by acblackford** [comment]
@anayeaye I sent @slesaad a copy of the jsons. I believe the datasets are ready for publication if those jsons have all of the necessary information.

File: .env
Comment (anayeaye) [inline .env:11]:
```suggestion
API_RASTER_ENDPOINT='https://openveda.cloud/api/raster'
```

File: .env
Comment (anayeaye) [inline .env:14]:
```suggestion
API_STAC_ENDPOINT='https://openveda.cloud/api/stac'
```

File: datasets/nlcd.data.mdx
Comment (anayeaye) [inline datasets/nlcd.data.mdx:40]:
```suggestion
    sourceParams:
        assets: landcover
        bidx: [1]
```

**Comment by anayeaye** [comment]
This PR is blocked by the need to add a custom colormap for the landcover data (https://github.com/NASA-IMPACT/veda-backend/issues/429). The firewall rejects a URL long enough to define the full NLCD colormap definition.

**Comment by slesaad** [comment]
image

the legend needs to be fixed cc @hanbyul-here @sandrahoang686

File: datasets/nlcd.data.mdx
Comment (slesaad) [inline datasets/nlcd.data.mdx:223]: These are internal links. Use `<Link>` component instead.

See example [here](https://github.com/US-GHG-Center/veda-config-ghg/pull/572/commits/f91b4041e359d15daea54e78b2fa01f4190c7af6)

File: datasets/nlcd-urbanization.data.mdx
Comment (slesaad) [inline datasets/nlcd-urbanization.data.mdx:179]: Internal link, use `<Link>` instead

can you find all of such URLs and update them. See example here
https://github.com/US-GHG-Center/veda-config-ghg/pull/572/commits/f91b4041e359d15daea54e78b2fa01f4190c7af6

Comment (acblackford) [reply datasets/nlcd-urbanization.data.mdx]: @slesaad I have updated the internal link with <Link> style, but the preview is failing due to an unexpected character error ',' . Is this because the story titles used in the link prose have commas?

**Comment by acblackford** [comment]
@snmln @hanbyul-here this needs released as soon as possible. It should have been included in the last release. My apologies if you were not made aware.

File: datasets/nlcd-urbanization.data.mdx
Comment (aboydnw) [inline datasets/nlcd-urbanization.data.mdx:25]: Should this be a production link now?

File: datasets/nlcd.data.mdx
Comment (aboydnw) [inline datasets/nlcd.data.mdx:24]: Should this be a production link now?

**Review by aboydnw — CHANGES_REQUESTED** [review]
I couldn't get the data to load for `nlcd-urbanization.data.mdx` or `nlcd.data.mdx`
Maybe that's related to the staging links? I think they should be to prod now, but let me know if I'm mistaken. So, at a minimum we'd want to make sure those are functional.

The other thing I might suggest is to consider making the two nlcd datasets two different layers under the same dataset. I think you could combine the dataset overview content into one file pretty easily, and the layers would still be available separately to reference in stories and visualize in E&A. Feel free to leave as-is if there is a strong reason for it, but I at least want to know that they are visualizing properly, pointing at the correct endpoints.

**Comment by acblackford** [comment]
> I couldn't get the data to load for `nlcd-urbanization.data.mdx` or `nlcd.data.mdx` Maybe that's related to the staging links? I think they should be to prod now, but let me know if I'm mistaken. So, at a minimum we'd want to make sure those are functional.
> 
> The other thing I might suggest is to consider making the two nlcd datasets two different layers under the same dataset. I think you could combine the dataset overview content into one file pretty easily, and the layers would still be available separately to reference in stories and visualize in E&A. Feel free to leave as-is if there is a strong reason for it, but I at least want to know that they are visualizing properly, pointing at the correct endpoints.

Per the second suggestion, it was originally set up like that but the collections had to be separate in STAC due to datetime + AOI overlaps so I split them in the dataset.mdx as well. I can combine the mdx though if that is best practice.

**Comment by aboydnw** [comment]
@acblackford The dashboard doesn't necessarily need to reflect the same organization in STAC, so I'd say merge them together in one dataset file if that makes the most sense for users!

@anayeaye we are having trouble visualizing these NLCD datasets, although the urbanization dataset that does not call on the custom colormap is working. Do you think this might be something to do with the colormap that was added [here](https://github.com/NASA-IMPACT/veda-backend/issues/429)?

**Comment by acblackford** [comment]
@aboydnw @anayeaye I have combined the dataset landing pages into one. The urbanization data is showing up fine but the multi-class NLCD dataset still is not populating.

File: datasets/nlcd.data.mdx
Comment (anayeaye) [inline datasets/nlcd.data.mdx:25]:
```suggestion
```
temporary staging url workaround no longer needed, we can use the default configured stac api and tiler

File: datasets/nlcd.data.mdx
Comment (anayeaye) [inline datasets/nlcd.data.mdx:102]:
```suggestion
```

File: datasets/aerosol-difference.data.mdx
Comment (anayeaye) [inline datasets/aerosol-difference.data.mdx:38]:
```suggestion
      datasetId: nlcd-annual-conus
```
because both layers are now in the nlcd-annual-conus mdx

**Review by anayeaye — COMMENTED** [review]
A few small changes after the dataset consolidation. I am still unclear on why the dashboard preview is looking for a cog_default in the nlcd annual conus explorer, though. I am wondering if we have a duplicated id somewhere causing trouble? nlcd properly renders the categorical colormap in openveda.cloud. I'll keep looking

File: datasets/nlcd.data.mdx
Comment (acblackford) [reply datasets/nlcd.data.mdx:25]: are you suggesting to remove these two lines entirely?

Comment (anayeaye) [reply datasets/nlcd.data.mdx]: yeah, I added them in this [PR](https://github.com/NASA-IMPACT/veda-config/pull/450/files) when we didn't have the nlcd custom colormap in production but now we do. Interestingly, the [nlcd-annual-conus preview ](https://deploy-preview-450--visex.netlify.app/exploration?search=&datasets=%5B%7B%22id%22%3A%22nlcd-annual-conus%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Average%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22St+Deviation%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%5D&taxonomy=%7B%7D&date=2021-01-01T07%3A00%3A00.000Z)for that PR is working.

**Pull request closed without merging**
